sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -350,7 +351,11 @@ private synchronized Connection getOrCreateConnection() throws IOException {
             connection = reactor.connectionToHost(handler.getHostname(), handler.getProtocolPort(), handler);
 
             reactorExceptionHandler = new ReactorExceptionHandler();
-            executor = new ReactorExecutor(reactor, Schedulers.single(), connectionId,
+            // Use a new single-threaded scheduler for this connection as QPID's Reactor is not thread-safe.
+            // Using Schedulers.single() will use the same thread for all connections in this process which
+            // limits the scalability of the no. of concurrent connections a single process can have.
+            Scheduler scheduler = Schedulers.newSingle("reactor-executor");
```

Comment:
Schedulers.single()  always create a daemon thread  and `newSingle` will create non-daemon thread. 

So in case of `newSingle` , Any existing user threads prevent the application from terminating.

Reply:
There is another API `Schedulers.newSingle(String name, boolean daemon)` that should be used in this case.

Reply:
We discussed this and we didn't want to make this a daemon thread.